src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,455 @@
+
+        // Get current exchange rate from oracle
+        uint256 exchangeRate = _oracle.getPriceForRedemption();
+        
+        // Calculate amounts using parent functionality
+        uint256 collateralAmount = calculateSaleReturn(depositAmount_);
+        uint256 feeAmount = (collateralAmount * sellFee) / BPS; // potentiallt wrong
+        uint256 redemptionAmount = collateralAmount - feeAmount;
+
+        // Generate new order ID
+        _orderId = _nextOrderId++;
+
+        // Create and store order
+        _redemptionOrders[_orderId] = RedemptionOrder({
+            seller: msg.sender,
+            sellAmount: depositAmount_,
+            exchangeRate: exchangeRate,
+            collateralAmount: collateralAmount,
+            feePercentage: sellFee,
+            feeAmount: feeAmount,
+            redemptionAmount: redemptionAmount,
+            collateralToken: address(token()),
+            redemptionTime: block.timestamp,
+            state: RedemptionState.Processing
+        });
+
+        // Update open redemption amount
+        openRedemptionAmount += redemptionAmount;
+
+
+        // Emit event with all order details
+        emit RedemptionOrderCreated(
+            _orderId,
+            msg.sender,
+            depositAmount_,
+            exchangeRate,
+            collateralAmount,
+            sellFee,
+            feeAmount,
+            redemptionAmount,
+            address(token()),
+            block.timestamp,
+            RedemptionState.Processing
+        );
+    }
+
+    /// @inheritdoc IFundingManager_v1
+    /// @param to_ The recipient address
+    /// @param amount_ The amount to transfer
+    function transferOrchestratorToken(
+        address to_,
+        uint256 amount_
+    ) external override(IFundingManager_v1) onlyPaymentClient {
+        token().safeTransfer(to_, amount_);
+
+        emit TransferOrchestratorToken(to_, amount_);
+    }
+
+    /// @notice Sets fee for sell operations
+    /// @param fee_ New fee amount
+    function setSellFee(
+        uint256 fee_
+    ) public override(RedeemingBondingCurveBase_v1, IRedeemingBondingCurveBase_v1) onlyOrchestratorAdmin {
+        _setSellFee(fee_);
+    }
+
+    function getSellFee() public pure returns (uint256 sellFee_) {
+        return sellFee_;
+    }
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    /// @param depositAmount_ The amount being deposited
+    /// @return mintAmount_ The calculated token amount
+    function _issueTokensFormulaWrapper(
+        uint256 depositAmount_
+    ) internal view override(BondingCurveBase_v1) returns (uint256 mintAmount_) {
+        // Calculate mint amount through oracle price
+        uint256 tokenAmount_ = _oracle.getPriceForIssuance() * depositAmount_;
+
+        // Convert mint amount to issuing token decimals
+        mintAmount_ = FM_BC_Tools._convertAmountToRequiredDecimal(
```

Comment:
I think the conversion has to happen before the calculation right?

Reply:
ouch, yes thank you!